satpy/readers/fci_l1c_nc.py

```diff
@@ -400,9 +412,20 @@ def _get_dataset_measurand(self, key, info=None):
     @cached_property
     def orbital_param(self):
         """Compute the orbital parameters for the current segment."""
-        actual_subsat_lon = float(np.nanmean(self._get_aux_data_lut_vector("subsatellite_longitude")))
-        actual_subsat_lat = float(np.nanmean(self._get_aux_data_lut_vector("subsatellite_latitude")))
-        actual_sat_alt = float(np.nanmean(self._get_aux_data_lut_vector("platform_altitude")))
+        if self.is_iqt:
+            actual_subsat_lon = float(self.get_and_cache_npxr("data/mtg_geos_projection/attr/"
+                                                              "longitude_of_projection_origin"))
+            actual_subsat_lat = 0.0
+            actual_sat_alt = float(self.get_and_cache_npxr("data/mtg_geos_projection/attr/perspective_point_height"))
+            logger.info("IQT data the following parameter is hardcoded "
+                        f" satellite_actual_latitude = {actual_subsat_lat} ,"
+                        " These parameters are taken from the projection's dictionary"
+                        f"satellite_actual_longitude = {actual_subsat_lon} ,"
+                        f"satellite_sat_alt = {actual_sat_alt}")
+        else:
+             actual_subsat_lon = float(np.nanmean(self._get_aux_data_lut_vector("subsatellite_longitude")))
+             actual_subsat_lat = float(np.nanmean(self._get_aux_data_lut_vector("subsatellite_latitude")))
+             actual_sat_alt = float(np.nanmean(self._get_aux_data_lut_vector("platform_altitude")))
```

Comment:
This function is getting pretty long, would you mind putting the bodies of each case in new functions?

Reply:
@ClementLaplace could you please fix this?

Reply:
It's done